jerry-core/ecma/builtin-objects/ecma-builtin-global.c

```diff
@@ -588,13 +588,10 @@ ecma_builtin_global_object_escape (lit_utf8_byte_t *input_start_p, /**< routine'
                                    lit_utf8_size_t input_size) /**< routine's first argument's
                                                                 *   string buffer's size */
 {
-  /*
-   * The escape routine has two major phases: first we compute
-   * the length of the output, then we encode the input.
-   */
   const lit_utf8_byte_t *input_curr_p = input_start_p;
   const lit_utf8_byte_t *input_end_p = input_start_p + input_size;
-  lit_utf8_size_t output_length = 0;
+  ecma_stringbuilder_t builder = ecma_stringbuilder_create ();
+  static lit_utf8_byte_t result_chars[3];
```

Comment:
Why is this static? Also, `URI_ENCODED_BYTE_SIZE` should be used for the length.